### PR TITLE
feat: add imported flag for wallet

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-4441 MF-4569

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/dashboard/src/API.tsx [hunk 1]
```diff
@@ -31,7 +31,12 @@ export interface PluginServices {
         hasPassword(): Promise<boolean>
         changePassword(oldPassword: string, newPassword: string): Promise<void>
         setPassword(newPassword: string): Promise<void>
-        recoverWalletFromMnemonic(name: string, mnemonic: string, derivationPath?: string): Promise<string>
+        recoverWalletFromMnemonic(
+            name: string,
+            mnemonic: string,
+            derivationPath?: string,
+            isImported?: boolean,
```
Comment (UncleBill) [inline packages/dashboard/src/API.tsx]: A wallet has only a set of mnemonic, can we recover wallets from another set of mnemonic words?

File: packages/dashboard/src/pages/CreateMaskWallet/AddDeriveWallet/index.tsx [hunk 2]
```diff
@@ -126,6 +127,7 @@ const AddDeriveWallet = memo(function AddDeriveWallet() {
                         `${walletName}${pathIndex}`,
                         mnemonic,
                         `${HD_PATH_WITHOUT_INDEX_ETHEREUM}/${pathIndex}`,
+                        false,
```
Comment (UncleBill) [inline packages/dashboard/src/pages/CreateMaskWallet/AddDeriveWallet/index.tsx]: I think only the core wallet manager is allowed to decide which wallet is imported or derived.

File: packages/dashboard/src/API.tsx (hunk 1 above)
Comment (nuanyang233) [reply packages/dashboard/src/API.tsx]: No, but we can recovery wallet with private key and store json. And this is just a flag.

File: packages/shared-base/src/types/Wallet.ts [hunk 3]
```diff
@@ -35,6 +35,8 @@ export interface Wallet {
     deployed?: boolean
     /** persona identifier */
     identifier?: string
+    /** flag indicating whether the wallet is imported */
```
Comment (guanbinrui) [inline packages/shared-base/src/types/Wallet.ts]: Please clarify here what kinds of wallets belong to `imported`.

**Review by guanbinrui — DISMISSED** [review]

**Pull request merged**